sql/sql_partition.cc

```diff
@@ -5929,6 +5929,32 @@ the generated partition syntax in a correct manner.
           *partition_changed= true;
         }
       }
+
+      // In case of PARTITION BY KEY(), check if primary key has changed
+      // System versioning also implicitly adds/removes primary key parts
+      if (alter_info->partition_flags == 0 && part_info->list_of_part_fields
+          && part_info->part_field_list.elements == 0 && !*partition_changed)
```

Comment:
SQL layer coding style recommends operators at the end of line contrary to InnoDB which recommends operator at the beginning of line. Please see code around if you have doubts how to format.

Reply:
where did you get that??

Reply:
well, ok, i think you're right